gslib/tests/util.py

```diff
@@ -172,6 +269,45 @@ def GetUserGroups():
 POSIX_INSUFFICIENT_ACCESS_ERROR = 'Insufficient access with uid/gid/mode for %s'
 
 
+class KmsTestingResources(object):
+  """Constants for KMS resource names to be used in integration testing."""
+  KEYRING_LOCATION = 'us-central1'
+  # Since KeyRings and their child resources cannot be deleted, we minimize the
+  # number of resources created by using a hard-coded keyRing name.
+  KEYRING_NAME = 'keyring-for-gsutil-integration-tests'
+
+  # Used by tests where we don't need to alter the state of a cryptoKey and/or
+  # its IAM policy bindings once it's initialized the first time.
+  CONSTANT_KEY_NAME = 'key-for-gsutil-integration-tests'
+  CONSTANT_KEY_NAME2 = 'key-for-gsutil-integration-tests2'
+
+  # This key should not be authorized so it can be used for failure cases.
+  CONSTANT_KEY_NAME_DO_NOT_AUTHORIZE = 'key-for-gsutil-no-auth'
+  # Pattern used for keys that should only be operated on by one tester at a
+  # time. Because multiple integration test invocations can run at the same
+  # time, we want to minimize the risk of them operating on each other's key,
+  # while also not creating too many one-time-use keys (as they cannot be
+  # deleted). Tests should fill in the %d entries with a digit between 0 and 9.
+  MUTABLE_KEY_NAME_TEMPLATE = 'cryptokey-for-gsutil-integration-tests-%d%d%d'
+
+
+def AuthorizeProjectToUseTestingKmsKey(
+    key_name=KmsTestingResources.CONSTANT_KEY_NAME):
+  """Ensures test keys exist and that the service agent is authorized."""
+  kms_api = KmsApi(logging.getLogger())
+
+  keyring_fqn = kms_api.CreateKeyRing(
```

Comment:
Might want to explain what `fqn` means somewhere

Reply:
fqn is now fully qualified.